cirq-core/cirq/ops/controlled_gate.py

```diff
@@ -161,34 +161,29 @@ def _decompose_with_context_(
         self, qubits: Tuple[cirq.Qid, ...], context: Optional[cirq.DecompositionContext] = None
     ) -> Union[None, NotImplementedType, cirq.OP_TREE]:
         control_qubits = list(qubits[: self.num_controls()])
-        if (
-            protocols.has_unitary(self.sub_gate)
-            and protocols.num_qubits(self.sub_gate) == 1
-            and self._qid_shape_() == (2,) * len(self._qid_shape_())
-            and isinstance(self.control_values, cv.ProductOfSums)
+        if protocols.has_unitary(self.sub_gate) and self._qid_shape_() == (2,) * len(
```

Comment:
While you're here, can you change the second condition to `all(q.dimension == 2 for q in qubits)` since it's equivalent and more readable?